Simplified/NYPLNetworkQueue.swift

```diff
@@ -1,41 +1,57 @@
 import Foundation
 import SQLite
 
+/**
+ The NetworkQueue is insantiated once on app startup and listens
+ for a valid network notification from a reachability class. It then
+ will retry any queued requests and purge them if necessary.
+ */
 final class NetworkQueue: NSObject {
-  
+
+  static let shared = NetworkQueue()
+
+  override init() {
+    super.init()
+    NotificationCenter.default.addObserver(forName: NSNotification.Name.NYPLReachabilityHostIsReachable,
```

Comment:
Even though this only gets used via a shared instance for now, you should probably still pair this with a `removeObserver` call in `deinit`.